lib/src/main/kotlin/io/provenance/core/KeyEntityManager.kt

```diff
@@ -6,8 +6,11 @@ class KeyEntityManager {
     private val entities: MutableMap<String, KeyEntity> = mutableMapOf()
     private val plugins: MutableSet<Plugin<*>> = mutableSetOf()
 
-    fun register(plugin: Plugin<*>) =
-        plugins.add(plugin)
+    fun <T: PluginConfig> register(plugin: Plugin<T>) {
+        if (plugins.filterIsInstance<Plugin<T>>().isEmpty()) {
+            plugins.add(plugin)
+        }
```

Comment:
Okay with silently ignoring if it's a duplicate?

Reply:
Ya, in the past the `set` took care of duplicates. Since introduction of the generic types, the set didn't pick up that the same item already existed. 

I did also try implementing `hashcode` override but that didnt seem to help either :sadpanda:. Kinda dumb fix but it works :yolo: